compiler/rustc_hir_typeck/src/fn_ctxt/mod.rs

```diff
@@ -121,7 +122,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         param_env: ty::ParamEnv<'tcx>,
         body_id: hir::HirId,
     ) -> FnCtxt<'a, 'tcx> {
-        let body_def_id = inh.tcx.hir().local_def_id(body_id);
+        let body_def_id = inh.tcx.hir().opt_local_def_id(body_id).unwrap_or(CRATE_DEF_ID);
```

Comment:
the `FnCtxt` should always use a correct `LocalDefId`. Please change `body_id` to already be a `LocalDefId`.

Reply:
Yes I was just discussing with @cjgillot regarding a ICE, and I found my code a little bit messy